compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2270,34 +2270,24 @@ public void visit(BLangCatch bLangCatch) {
 
     @Override
     public void visit(BLangTransaction transactionNode) {
-        analyzeStmt(transactionNode.transactionBody, env);
+        SymbolEnv transactionEnv = SymbolEnv.createTransactionEnv(transactionNode, env);
```

Comment:
Just a questions: why do we need to create a new symbol environment here?

Reply:
So that we can identify transactionNode type in Typechecker when the variable is visited.

Please feel free to suggest if there is any other better way to do this, that i might be missing.

Reply:
ack